browsers/chrome/manifest.json

```diff
@@ -41,7 +41,12 @@
         "all_frames": true,
         "css": ["public/css/noatb.css"],
         "run_at": "document_start"
-      }
+      },
+      {
+        "matches": ["https://duckduckgo.com/*"],
+        "js": ["public/js/content-scripts/serp-document-start-handler.js"],
+        "run_at": "document_start"
```

Comment:
I'm not stoked that this script will always be injected (we need it once in most cases), but I understand that alternative of injecting it programmatically had timing issues?

Reply:
Yes not a fan either. I did that as manifest + `run_at: document_start` is the most reliable way to inject a script as early as possible in the page lifecycle (`onBeforeNavigate` is too early).
That being said thinking more I think that `onCommitted` + `document_start` should be good enough for our case so I went ahead with that change in this [commit](https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/568/commits/f209a89f1095fbec630d045a163c031bf945e65f). We lose a few ms but it's cleaner/less wasteful.